### PR TITLE
Improve integration tests 3

### LINKED ISSUE #67621: Flaky tests `test_storage_hdfs`
https://play.clickhouse.com/play?user=play#c2VsZWN0IAp0b1N0YXJ0T2ZEYXkoY2hlY2tfc3RhcnRfdGltZSkgYXMgZCwKY291bnQoKSwgIGdyb3VwVW5pcUFycmF5KHB1bGxfcmVxdWVzdF9udW1iZXIpLCAgYW55KHJlcG9ydF91cmwpCmZyb20gY2hlY2tzIHdoZXJlICcyMDI0LTAxLTAxJyA8PSBjaGVja19zdGFydF90aW1lIGFuZCB0ZXN0X25hbWUgbGlrZSAnJXRlc3Rfc3RvcmFnZV9oZGZzJScgYW5kIHRlc3Rfc3RhdHVzIGluICgnRkFJTCcsICdGTEFLWScsICdFUlJPUicpIGdyb3VwIGJ5IGQgb3JkZXIgYnkgZCBkZXNj

Maybe #66867 will help

### PR DESCRIPTION
**Pull request by qoega**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Closes #67621

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit 6573482f9f030e93773c6f52cac15f9e79cd0dbe with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/67843/6573482f9f030e93773c6f52cac15f9e79cd0dbe/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Fast test</td><td>Normally this is the first check that is ran for a PR. It builds ClickHouse and runs most of <a href="https://clickhouse.com/docs/en/development/tests#functional-tests">stateless functional tests</a>, omitting some. If it fails, further checks are not started until it is fixed. Look at the report to see which tests fail, then reproduce the failure locally as described <a href="https://clickhouse.com/docs/en/development/tests#functional-test-locally">here</a></td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/c5cf1ff57e1e50a8e9ed7cb100afbf30e8a3224c/fast_test.html">✅ success</a></td></tr>
<tr><td>Unit tests</td><td>Runs the unit tests for different release types</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/c5cf1ff57e1e50a8e9ed7cb100afbf30e8a3224c/unit_tests__release_.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: tests/integration/compose/docker_compose_hdfs.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 version: '2.3'
 services:
     hdfs1:
-        image: sequenceiq/hadoop-docker:2.7.0
+        image: prasanthj/docker-hadoop:2.6.0
```
Comment (antaljanosbenjamin) [inline tests/integration/compose/docker_compose_hdfs.yml:4]: Could you please explain this?

File: tests/integration/test_access_for_functions/test.py [hunk 2]
```diff
@@ -76,4 +76,9 @@ def test_ignore_obsolete_grant_on_database():
     )
     instance.start_clickhouse()
 
-    assert instance.query("SHOW GRANTS FOR X") == "GRANT SELECT ON mydb.* TO X\n"
+    assert (
+        instance.query(f"SHOW GRANTS FOR `{user_id}`")
+        == f"GRANT SELECT ON mydb.* TO `{user_id}`\n"
+    )
+    instance.stop_clickhouse()
+    instance.start_clickhouse()
```
Comment (antaljanosbenjamin) [inline tests/integration/test_access_for_functions/test.py:84]: Why do we need to stop/start clickhouse here?

File: tests/integration/compose/docker_compose_hdfs.yml (hunk 1 above)
Comment (qoega) [reply tests/integration/compose/docker_compose_hdfs.yml:4]: Old sequenceiq image has unsupported container image and we get warnings about deprecation. This was found here https://github.com/ClickHouse/ClickHouse/pull/66867 but not all tests work with new docker as we have changed network behaviour (different IPs)

File: tests/integration/test_access_for_functions/test.py (hunk 2 above)
Comment (qoega) [reply tests/integration/test_access_for_functions/test.py:84]: Not necessary now as I started to use different user names instead.

**Review by antaljanosbenjamin — APPROVED** [review]

File: tests/integration/test_access_for_functions/test.py (hunk 2 above)
Comment (qoega) [reply tests/integration/test_access_for_functions/test.py:84]: I can do it as a follow up PR. I just do not want to trigger the whole CI for this trivial change as other PR is waiting.

Comment (qoega) [reply tests/integration/test_access_for_functions/test.py:84]: https://github.com/ClickHouse/ClickHouse/pull/68007

**Comment by ilejn** [comment]
Hello @qoega , do you have a good reason to prefer prasanthj/docker-hadoop:2.6.0 over apache/hadoop:2.10.2 ?
Asking because I plan to refresh base image for https://github.com/ClickHouse/ClickHouse/blob/master/docker/test/integration/kerberized_hadoop/Dockerfile
Seems natural to use the same one.

**Comment by qoega** [comment]
Yes, because the old one does not work at all in recent docker as format is fully deprecated and OS is EOL long time ago. So this one is basically the same, but on top of Ubuntu and it works in docker

**Comment by ilejn** [comment]
> Yes, because the old one does not work at all in recent docker as format is fully deprecated and OS is EOL long time ago. So this one is basically the same, but on top of Ubuntu and it works in docker

Is seems that you explained why you prefer prasanthj/docker-hadoop:2.6.0 over sequenceiq/hadoop-docker:2.7.0 .
I am asking about apache/hadoop:2.10.2 .

**Comment by qoega** [comment]
I guess I checked and it is not all-in-one image if I recall correctly. I just wanted to have the same working and not rebuild the whole Hadoop test infrastructure

**Comment by ilejn** [comment]
@qoega , have a look at https://github.com/ClickHouse/ClickHouse/pull/70244 , please.

**Comment by maxknv** [comment]
we need updated hadoop image on release branches. I'll try to backport it

**Pull request merged**